app/models/fee.rb

```diff
@@ -5,6 +5,7 @@ class Fee < ApplicationRecord
 
   belongs_to :invoice
   belongs_to :charge, optional: true
+  belongs_to :add_on, optional: true
```

Comment:
Actually, I think the relation should be at the applied_add_on level. A bit like it's done for the coupon with the credit model.

Sorry it wasn't clear in my previous review :/

Reply:
update is pushed:)